source/includes/_oauth2.md

```diff
@@ -42,6 +42,32 @@ In this page they can grant access to a data partner by selecting the data partn
 
 <img src="images/authorization_flow/participant-granted-data-partner-access.png" alt="granted access">
 
+## OAuth flow
+
+An OAuth application can be created through your data partner account. Subsequently OAuth can be used to ask a user to grant access. This will then provide access through OAuth for the entities the user has access to. This will only work if the entities are also connected to your data partner account (see previous section). This is meant to secure user permission and saves you the exercise of mapping entities to your own internal platform. Since you can simply show to the user what is returned through OAuth.
+
+An OAuth application can be created through your data partner account.
+
+<img src="images/authorization_flow/oauth-data-partner.png" alt="oauth data partner">
+
+While creating a new OAuth application you need to add a redirect uri to which responses are send back.
+
+<img src="images/authorization_flow/oauth-create-new-application.png" alt="oauth new application">
+
+After creation you will see the application id and secret which you can hand in for a bearer token.
+
+<img src="images/authorization_flow/oauth-application-id-secret.png" alt="oauth id secret">
+
+{write here URL to request bearer token ask Bart and provide code blob in side bar}
```

Comment:
Probably something with `/oauth/token` and the paramaters `client_id` `client_secret`?

Reply:
Would suggest we return https://github.com/GRESB/api-docs/pull/126/files#diff-0a7ac57016342285b2391931393f18d6fbf17b7a7b859a0f3e69d551ff4db2e1L22. If it still works the same.

Reply:
it still works the same so I think that makes sense. After reverting that we might tweak it a little bit to contain the information about not only having to grant access via OAuth but also granting the partner access via the portal UI.

All the URLs related to OAuth are still the same with the exception of the one where a data partner creates their application. That changed from `/oauth/applications/new` to `/oauth/data_partners/<partner_id>/applications/new`

Reply:
and the `partner_id` comes from the Portal URL? For example `/data_partners/4/tokens`.

we don't return a `partner_id` in any endpoint as far as I'm aware.